File: python/src/Construct/LNEnums.cpp
/*
 * Author:
 * 2025/11/23 - Yuqing Liang (BIMCoder Liang)
 * bim.frankliang@foxmail.com
 * 
 *
 * Use of this source code is governed by a LGPL-2.1 license that can be found in
 * the LICENSE file.
 */

#include <pybind11/pybind11.h>
#include "LNEnums.h"

namespace py = pybind11;
void cstrEnums(py::module_&m)
{
    py::enum_<LNLib::CurveCurveIntersectionType>(m, "CurveCurveIntersectionType")
        .value("Intersecting", LNLib::CurveCurveIntersectionType::Intersecting)
        .value("Parallel", LNLib::CurveCurveIntersectionType::Parallel)
        .value("Coincident", LNLib::CurveCurveIntersectionType::Coincident)
        .value("Skew", LNLib::CurveCurveIntersectionType::Skew)
        .export_values();

    py::enum_<LNLib::LinePlaneIntersectionType>(m, "LinePlaneIntersectionType")
        .value("Intersecting", LNLib::LinePlaneIntersectionType::Intersecting)
        .value("Parallel", LNLib::LinePlaneIntersectionType::Parallel)
        .value("On", LNLib::LinePlaneIntersectionType::On)
        .export_values();

    py::enum_<LNLib::CurveNormal>(m, "CurveNormal")
        .value("Normal", LNLib::CurveNormal::Normal)
        .value("Binormal", LNLib::CurveNormal::Binormal)
        .export_values();

    py::enum_<LNLib::SurfaceDirection>(m, "SurfaceDirection")
        .value("All", LNLib::SurfaceDirection::All)
        .value("UDirection", LNLib::SurfaceDirection::UDirection)
        .value("VDirection", LNLib::SurfaceDirection::VDirection)
        .export_values();

    py::enum_<LNLib::SurfaceCurvature>(m, "SurfaceCurvature")
        .value("Maximum", LNLib::SurfaceCurvature::Maximum)
        .value("Minimum", LNLib::SurfaceCurvature::Minimum)
        .value("Gauss", LNLib::SurfaceCurvature::Gauss)
        .value("Mean", LNLib::SurfaceCurvature::Mean)
        .value("Abs", LNLib::SurfaceCurvature::Abs)
        .value("Rms", LNLib::SurfaceCurvature::Rms)
        .export_values();

    py::enum_<LNLib::IntegratorType>(m, "IntegratorType")
        .value("Simpson", LNLib::IntegratorType::Simpson)
        .value("GaussLegendre", LNLib::IntegratorType::GaussLegendre)
        .value("Chebyshev", LNLib::IntegratorType::Chebyshev)
        .export_values();

    py::enum_<LNLib::OffsetType>(m, "OffsetType")
        .value("TillerAndHanson", LNLib::OffsetType::TillerAndHanson)
        .value("PieglAndTiller", LNLib::OffsetType::PieglAndTiller)
        .export_values();

    py::enum_<LNLib::ExtensionType>(m, "ExtensionType")
        .value("Tangent", LNLib::ExtensionType::Tangent)
        .value("Arc", LNLib::ExtensionType::Arc)
        .value("Natural", LNLib::ExtensionType::Natural)
        .export_values();

    py::enum_<LNLib::VolumeDirection>(m, "VolumeDirection")
        .value("U", LNLib::VolumeDirection::U)
        .value("V", LNLib::VolumeDirection::V)
        .value("W", LNLib::VolumeDirection::W)
        .export_values();
}

File: dotnet/PInvoke/src/LNEnums.cs
/*
 * Author:
 * 2025/11/30 - Yuqing Liang (BIMCoder Liang)
 * bim.frankliang@foxmail.com
 * 
 *
 * Use of this source code is governed by a LGPL-2.1 license that can be found in
 * the LICENSE file.
 */

namespace LNLibSharp
{
public enum CurveCurveIntersectionType
{
    CURVE_CURVE_INTERSECTING = 0,
    CURVE_CURVE_PARALLEL = 1,
    CURVE_CURVE_COINCIDENT = 2,
    CURVE_CURVE_SKEW = 3
}

public enum LinePlaneIntersectionType
{
    LINE_PLANE_INTERSECTING = 0,
    LINE_PLANE_PARALLEL = 1,
    LINE_PLANE_ON = 2
}

public enum CurveNormal
{
    CURVE_NORMAL_NORMAL = 0,
    CURVE_NORMAL_BINORMAL = 1
}

public enum SurfaceDirection
{
    SURFACE_DIRECTION_ALL = 0,
    SURFACE_DIRECTION_U = 1,
    SURFACE_DIRECTION_V = 2
}

public enum SurfaceCurvature
{
    SURFACE_CURVATURE_MAXIMUM = 0,
    SURFACE_CURVATURE_MINIMUM = 1,
    SURFACE_CURVATURE_GAUSS = 2,
    SURFACE_CURVATURE_MEAN = 3,
    SURFACE_CURVATURE_ABS = 4,
    SURFACE_CURVATURE_RMS = 5
}

public enum IntegratorType
{
    INTEGRATOR_SIMPSON = 0,
    INTEGRATOR_GAUSS_LEGENDRE = 1,
    INTEGRATOR_CHEBYSHEV = 2
}

public enum OffsetType
{
    OFFSET_TILLER_HANSON = 0,
    OFFSET_PIEGL_TILLER = 1
}
}